content/blog/teach-tidyverse-2021/index.Rmd

```diff
@@ -0,0 +1,437 @@
+
+```{r eval = FALSE}
+read_csv(files, id = "path")
+#> # A tibble: 7 × 3
+#>   path            brand     n
+#>   <chr>           <dbl> <dbl>
+#> 1 data/sales-aug…  1234     8
+#> 2 data/sales-aug…  8721     2
+#> 3 data/sales-aug…  1822     3
+#> 4 data/sales-sep…  3333     1
+#> 5 data/sales-sep…  2156     3
+#> 6 data/sales-sep…  3987     6
+#> 7 data/sales-sep…  3216     5
+```
+
+Previously this not-so-advanced task required the use of mapping functions from purrr or the [vroom](https://vroom.r-lib.org/) package, but now tidyverse users are able to accomplish this task with just readr!
+
+## Web scraping with rvest
+
+If you've been teaching web scraping with [**rvest**](https://rvest.tidyverse.org/), I recommend updating your teaching materials as you might be able to further simplify and streamline some of the code you present to students.
+And if you haven't been teaching web scraping, I recommend reading our paper titled [Web Scraping in the Statistics and Data Science Curriculum: Challenges and Opportunities](https://www.tandfonline.com/doi/full/10.1080/10691898.2020.1787116) where we discuss how web scraping can be implemented in a pedagogically sound and technically executable way at various levels of statistics and data science curricula.
+
+Most recent updates to rvest include the addition of a new function, `html_text2()`, which offers better handling for line breaks.
+Suppose you have the following paragraph of text across two lines on a webpage.
+
+```{r}
+library(rvest)
+
+html <- minimal_html(
+  "<p>  
+    This is the first sentence in the paragraph.
+    This is the second sentence that should be on the same line as the first sentence.<br>This third sentence should start on a new line.
+  </p>"
+)
+```
+
+With the original `html_text()` function extracting the text out of this paragraph results in the following:
+
+```{r}
+html %>% html_text() %>% writeLines()
+```
+
+Note that the line breaks in the output do not respect the line break defined with `<br>`.
+
+With the new `html_text2()`, `<br>` is handled appropriately and the line breaks follow the expected pattern.
+
+```{r}
+html %>% html_text2() %>% writeLines()
+```
+
+The output of `html_text2()` is generally what you want, but note that it is slower than `html_text()`.
+This might not make a big difference for teaching web scraping as a new topic, but it is worth keeping in mind when the task involves scraping a large amount of data.
+Your choice might also depend on what you're going to do next with the data.
+For example, if the next step involves tokenizing the scraped text with `tidytext::unnest_tokens()` you might not care how the line breaks were handled in the first step.
+
+Since this change involves the addition of a new function without changing behaviour in any existing functions, incorporating it into your teaching would require testing `html_text2()` in places where you previously used `html_text()` to see if the result is preferable.
+
```

Comment:
I think you can still mention `html_table()` here since it should be generally better even if that one example I provided is wrong 😬. Maybe something like: If `html_table()` didn't work for you in the past, it's worth trying again since it's been rewritten from scratch to more closely match how browsers display tables with merged cells.